components/http-accept.js

```diff
@@ -19,89 +18,58 @@ exports.getComponent = componentFactory({
             description: "HTTP 202 Accepted request/response pair {req, res}",
             datatype: 'object'
         }
-    }, promiseOutput.outPorts),
-    inPorts: {
-        limit: {
-            description: "Request Content-Length limit",
-            datatype: 'int',
-            ondata: function(limit) {
-                this.nodeInstance.limit = limit;
-            }
-        },
-        encoding: {
-            description: "Request body character encoding",
-            datatype: 'string',
-            ondata: function(encoding) {
-                this.nodeInstance.encoding = encoding;
-            }
-        },
-        type: {
-            description: "Request Content-Type",
-            datatype: 'string',
-            ondata: function(type) {
-                this.nodeInstance.types = this.nodeInstance.types || [];
-                this.nodeInstance.types.push(type);
-            }
-        },
-        input: {
-            description: "HTTP request/response pair {req, res}",
-            datatype: 'object',
-            required: true,
-            ondata: promiseOutput(handle)
-        }
-    }
-});
-
-function handle(pair){
-    var self = this.nodeInstance;
-    var outPorts = this.nodeInstance.outPorts;
-    if (contentTypeMatches(self.types, pair.req.headers['content-type'])) {
-        if (_.has(pair.req, 'body')) {
-            if (outPorts.accepted.listAttached().length) {
-                pair.res.writeHead(202, "Accepted");
-                pair.res.write("Accepted");
-                outPorts.accepted.send(pair);
-                outPorts.accepted.disconnect();
-            }
-            return pair.req.body;
-        } else {
-            var charset = pair.req.headers['content-type'] &&
-                typer.parse(pair.req.headers['content-type']).parameters.charset;
-            return getRawBody(pair.req, {
-                limit: self.limit || '1mb',
-                encoding: charset || self.encoding || 'utf8'
-            }).then(function(body){
+    },
+    updater: function(limit, encoding, type, input) {
+        var outPorts = this.nodeInstance.outPorts;
+        if (contentTypeMatches(type, input.req.headers['content-type'])) {
+            if (_.has(input.req, 'body')) {
                 if (outPorts.accepted.listAttached().length) {
-                    pair.res.writeHead(202, "Accepted");
-                    pair.res.write("Accepted\n");
-                    outPorts.accepted.send(pair);
+                    input.res.writeHead(202, "Accepted");
+                    input.res.write("Accepted");
+                    outPorts.accepted.send(input);
```

Comment:
If possible we should get away from noflo-specific calls.  Is this needed?  Could we use a shim node instead?